src/main/java/jenkins/branch/MultiBranchProjectViewHolder.java

```diff
@@ -233,7 +234,10 @@ public ViewImpl(ViewGroup owner, @NonNull SCMHeadCategory category) {
          */
         @Override
         public String getDisplayName() {
-            return category.getDisplayName().toString() + " (" + getItems().size() + ")";
+            //Only used to get a count and to prevent poor performance in cases with large number of items
+            try (ACLContext ctx = ACL.as(ACL.SYSTEM)) {
+                return category.getDisplayName().toString() + " (" + getItems().size() + ")";
```

Comment:
I guess the drawback is that if someone doesn't have permission to view all of the items, the number here will be wrong? I wonder how common it is for users to configure different permissions for different jobs in a multibranch project or different multibranch projects in an organization folder. My guess is that it's not super common, so I'm not too worried about it, although I do wonder if some users would consider that a problem in terms of information disclosure if some of the jobs are supposed to be hidden from non-admins or similar.

Reply:
Yes the number would be wrong, I'm not sure what scenarios would read permissions be different especially for a github repo(private or public) since I don't think there is a way to give per branch permissions in github.

Currently I'm not sure of any way around this except maybe editing core where the number is handled in jelly. But this doesn't help the scenario when the views are sorted.

https://github.com/jenkinsci/jenkins/blob/34f5ea55cb77b83da9dc0529f597f2948d425ae4/core/src/main/resources/hudson/views/DefaultViewsTabBar/viewTabs.jelly#L29-L31

Reply:
@res0nance 
It's not just that the number would be wrong as Devin said - it gives the user information they shouldn't have.